code/game/machinery/deployable.dm

```diff
@@ -210,6 +223,202 @@
+	if(core_shield)
+		source.protected = FALSE
+	return ..()
+
+/obj/structure/barricade/dropwall/emp_act(severity)
+	..()
+	obj_integrity -= 40 / severity //chances are the EMP will also hit the generator, we don't want it to double up too heavily
+
+/obj/structure/barricade/dropwall/bullet_act(obj/item/projectile/P)
+	if(P.shield_buster)
+		qdel(src)
+	else
+		. = ..()
+
+/obj/item/grenade/barrier/dropwall
+	name = "dropwall shield generator"
+	desc = "This generator designed by Shellguard Munitions's spartan division is used to deploy a temporary cover that blocks projectiles and explosions from a direction, while allowing projectiles to pass freely from behind."
+	actions_types = list(/datum/action/item_action/toggle_barrier_spread)
+	icon = 'icons/obj/dropwall.dmi'
+	icon_state = "dropwall"
+	mode = NORTH
+	var/uptime = 12 SECONDS
+
+/obj/item/grenade/barrier/dropwall/toggle_mode(mob/user)
+	switch(mode)
+		if(NORTH)
+			mode = EAST
+		if(EAST)
+			mode = SOUTH
+		if(SOUTH)
+			mode = WEST
+		if(WEST)
+			mode = NORTH
+
+	to_chat(user, "[src] is now in [dir2text(mode)] mode.")
+
+/obj/item/grenade/barrier/dropwall/prime()
+	new /obj/structure/dropwall_generator(get_turf(loc), mode, uptime)
+	qdel(src)
+
+/obj/structure/dropwall_generator
+	name = "deployed dropwall shield generator"
+	desc = "This generator designed by Shellguard Munitions's spartan division is used to deploy a temporary cover that blocks projectiles and explosions from a direction, while allowing projectiles to pass freely from behind."
+	icon = 'icons/obj/dropwall.dmi'
+	icon_state = "dropwall_deployed"
+	max_integrity = 25 // 2 shots
+	var/list/connected_shields = list()
+	/// This variable is used to prevent damage to it's core shield when it is up.
+	var/protected = FALSE
+	///The core shield that protects the generator
+	var/obj/structure/barricade/dropwall/core_shield = null
+
+/obj/structure/dropwall_generator/Initialize(mapload, direction, uptime)
+	. = ..()
+	if(direction)
+		deploy(direction, uptime)
+
+/obj/structure/dropwall_generator/proc/deploy(direction, uptime)
+	anchored = TRUE
+	protected = TRUE
+	addtimer(CALLBACK(src, .proc/power_out), uptime)
+	while(uptime >= 10)
+		addtimer(CALLBACK(src, .proc/annoying_overlay_proc, uptime / 10), (10 * (12 - uptime / 10))) // instantly, 1 second, 2 seconds, ect. Would put in SECONDS here but it shits itself
+		uptime -= 10
+
+	switch(direction)
+		if(NORTH)
+			connected_shields += new /obj/structure/barricade/dropwall(get_turf(loc), src, TRUE, NORTH)
+			for(var/i in connected_shields)
+				core_shield = i
+
+			var/target_turf = get_step(src, EAST)
+			if(!(is_blocked_turf(target_turf)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf, src, FALSE, NORTH, EAST)
+
+			var/target_turf2 = get_step(src, WEST)
+			if(!(is_blocked_turf(target_turf2)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf2, src, FALSE, NORTH, WEST)
+		if(EAST)
```

Comment:
instead of duplicating this code multiple times and just changing the direction, you could use the `turn` function to do something like `var/dir_left = turn(direction, -90)` and `var/dir_right = turn(direction, 90)`

Reply:
Done!